File: Chapter-2/2/count.pas
{ fpc count.pas }
program Count;

var
  sentence: string;
  sentence_length: integer;

begin
  write('What is the input string? ');
  readln(sentence);
  sentence_length := length(sentence);

  if sentence_length = 0
    then writeln('Empty string!')
    else writeln(sentence, ' has ', length(sentence), ' characters.');
end.

File: Chapter-2/2/count.cpp
#include <iostream>
#include <string>
using namespace std;

int main() {
  cout << "What is the input string? ";
  string s;
  getline(cin, s);
  if (s.length() == 0) {
    cout << "Empty string!" << endl;
  } else {
    cout << s << " has " << s.length() << " characters." << endl;
  }
  return 0;
}
